ethers-contract/ethers-contract-abigen/src/multi.rs

```diff
@@ -570,22 +570,36 @@ impl MultiBindingsInner {
     }
 
     /// parses the active Cargo.toml to get what version of ethers we are using
-    fn find_crate_version(&self) -> Result<String> {
+    fn find_crate_version(&self, lib: &Path) -> Result<String> {
         let cargo_dir = Path::new(env!("CARGO_MANIFEST_DIR")).join("Cargo.toml");
         let data = std::fs::read_to_string(cargo_dir)?;
         let toml = data.parse::<Value>()?;
 
-        let Some(Some(ethers)) = toml.get("dependencies").map(|v| v.get("ethers")) else { eyre::bail!("couldn't find ethers dep")};
+        let Some(ethers) = toml.get("dependencies")
+            .and_then (|v| v.get("ethers").or_else(|| v.get("ethers-contract")))
+            else { eyre::bail!("couldn't find ethers or ethers-contract dependency")};
         if ethers.is_table() {
             if let Some(rev) = ethers.get("rev") {
-                return Ok(format!("ethers = {{ git = \"https://github.com/gakonst/ethers-rs\", rev = {}, default-features = false, features = [\"abigen\"] }}", rev));
+                Ok(format!("ethers = {{ git = \"https://github.com/gakonst/ethers-rs\", rev = {}, default-features = false, features = [\"abigen\"] }}", rev))
             } else if let Some(version) = ethers.get("version") {
-                return Ok(format!("ethers = {{ version = {}, default-features = false, features = [\"abigen\"] }}", version ));
+                if let Some(path) = ethers.get("path") {
+                    let path = Path::new(path.type_str());
+                    let mut res =
+                        lib.to_str().unwrap().split("//").fold(PathBuf::from(""), |mut acc, _| {
+                            acc.push("../");
+                            acc
+                        });
+                    res.extend(path);
+
+                    Ok(format!("ethers = {{ version = {}, path = {} default-features = false, features = [\"abigen\"] }}", version, res.to_string_lossy()))
+                } else {
+                    Ok(format!("ethers = {{ version = {}, default-features = false, features = [\"abigen\"] }}", version ))
+                }
             } else {
-                return Ok("ethers = {{ git = \"https://github.com/gakonst/ethers-rs\", default-features = false, features = [\"abigen\"] }}".to_string());
+                Ok("ethers = {{ git = \"https://github.com/gakonst/ethers-rs\", default-features = false, features = [\"abigen\"] }}".to_string())
             }
```

Comment:
I'm sorry I made a mistake in the last review, it should of course be `git` not `path`.

imo this can all be replaced by checking only for the existence of `git`. I wouldn't bother with the `path` at all. 

so the condition should just be `if ethers.get("git").is_some()`

Reply:
Right because abigen is a feature so will always be a table.

Reply:
> so the condition should just be `if ethers.get("git").is_some()`
I don't believe we can just check for `git` though as another possible flag is `rev` which specifies a specific git hash. I think the best way to do this would be the way that the code I pushed currently handles it.

Reply:
@mattsse thoughts on above?

Reply:
rev requires git tough: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-git-repositories

Reply:
> rev requires git tough

yes I'm aware. I think its a good idea to allow specification of a specific hash due to dependency conflicts with other crates and I've actually had to use this a few times where the most recent version had a bug i wanted to avoid but the most recent git verision gave me a dep conflict. It's a pretty niche case but I think its more correct overall. Thinking o

If you think its a bit overkill I can remove it, just lmk.

Reply:
i think OK as-is